lib/utilities/clean-remove.js

```diff
@@ -4,45 +4,41 @@ const path = require('path');
 const fs = require('fs-extra');
 const walkUp = require('./walk-up-path');
 
-function cleanRemove(fileInfo) {
-  // see if file exists to avoid wasting time
-  return fs
-    .stat(fileInfo.outputPath)
-    .then(() => fs.remove(fileInfo.outputPath))
-    .then(() => {
-      let paths = walkUp(fileInfo.displayPath).map((thePath) => path.join(fileInfo.outputBasePath, thePath));
+async function cleanRemove(fileInfo) {
+  try {
+    await fs.stat(fileInfo.outputPath);
+    await fs.remove(fileInfo.outputPath);
+    let paths = walkUp(fileInfo.displayPath).map((thePath) => path.join(fileInfo.outputBasePath, thePath));
 
-      return paths.reduce(
-        (chainedPromise, thePath) =>
-          chainedPromise.then((wasShortCircuited) => {
-            if (wasShortCircuited) {
-              // optimization that says since my child dir wasn't empty,
-              // I can't be empty, so keep skipping
-              return true;
-            }
+    return paths.reduce(
+      (chainedPromise, thePath) =>
+        chainedPromise.then(async (wasShortCircuited) => {
```

Comment:
Can we kill this reduce?

AFAICT this whole thing can be boiled down to this (which I think is much simpler):

```js
for (let thePath of paths) {
  try {
    let childPaths = await fs.readdir(thePath);
    if (childPaths.length > 0) {
      return;
    }
  
    await fs.remove(thePath);
  } catch(err) {
    if (err.code === 'ENOENT') {
      // do nothing, already removed
    }

    throw err;
  }
}
```